usaspending_api/common/api_request_utils.py

```diff
@@ -464,12 +466,16 @@ def build_response(self):
                     "matched_ids": Location.objects.filter(Q(**{"location_congressional_code": row[0], "location_state_code": row[1], "location_state_name": row[2]})).values_list("location_id", flat=True)
                 }
                 response_object.append(response_row)
+                if len(response_object) >= limit:
```

Comment:
Thanks, @afrasier. As I look at this, I keep getting the feeling that we could go even deeper re: what we can accomplish in the query vs having to manually spin through its results. 

For example, do you think it would be possible to annotate the queries on lines 457 and 472 to return `place`, `place_type`, and `parent` and then do an additional join to pull in the `matched_ids`? And just serialize/return a response within the confines of DRF instead of crafting a bespoke response?

This is a high-level spitball, and maybe it wouldn't work. But would be a good exercise to think through as we work on our endpoints that don't inherit from the ViewSets.